src/Castle.Core/DynamicProxy/Contributors/ClassProxyWithTargetTargetContributor.cs

```diff
@@ -141,17 +149,19 @@ private Type GetInvocationType(MetaMethod method, ClassEmitter @class)
 		}
 
 		private MethodGenerator IndirectlyCalledMethodGenerator(MetaMethod method, ClassEmitter proxy,
-		                                                        OverrideMethodDelegate overrideMethod)
+		                                                        OverrideMethodDelegate overrideMethod,
+		                                                        bool skipInterceptors = false)
 		{
 			var @delegate = GetDelegateType(method, proxy);
 			var contributor = GetContributor(@delegate, method);
 			var invocation = new CompositionInvocationTypeGenerator(targetType, method, null, false, contributor)
 				.Generate(proxy, namingScope)
 				.BuildType();
 			return new MethodWithInvocationGenerator(method,
-			                                         proxy.GetField("__interceptors"),
+			                                         skipInterceptors ? NullExpression.Instance : proxy.GetField("__interceptors").ToExpression(),
```

Comment:
TBH this may be a bit of a hack: It relies on `AbstractInvocation.Proceed` calling `InvokeMethodOnTarget` right away when `interceptors == null` ([see here](https://github.com/castleproject/Core/blob/d1310e06b52dba1f7d5a35fd4d47482878460fd0/src/Castle.Core/DynamicProxy/AbstractInvocation.cs#L102-L107)), so we get the needed behavior without having to implement a new type of `MethodGenerator`. But the generated proxied method body in this case perhaps does too muchwork, and could be optimized further.

At this time, I don't yet understand the `MethodGenerator` part of DP well enough to add a new variant. I may pick this up in the future, when the code base has been cleaned up a little more and when I'm more familiar with it. I'm hoping this will do in the meantime.